src/Http/Offer/PriceInfoValidatingRequestBodyParser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Offer;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use CultuurNet\UDB3\Json;
+use CultuurNet\UDB3\Model\Serializer\ValueObject\Price\PriceInfoDenormalizer;
+use CultuurNet\UDB3\Model\ValueObject\Price\PriceInfo;
+use Psr\Http\Message\ServerRequestInterface;
+
+class PriceInfoValidatingRequestBodyParser implements RequestBodyParser
+{
+    public function parse(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $data = $request->getParsedBody();
+        if (!isset($data->priceInfo) || !is_array($data->priceInfo)) {
+            return $request;
+        }
+
+        $priceInfoDenormalizer = new PriceInfoDenormalizer();
+        $priceInfo = $priceInfoDenormalizer->denormalize(
+            Json::decodeAssociatively(Json::encode($data->priceInfo)),
+            PriceInfo::class
+        );
+
+        if ($priceInfo->getTariffs()->hasDuplicates()) {
+            throw ApiProblem::bodyInvalidData(
+                new SchemaError(
+                    '/priceInfo',
+                    'Tariff names should be unique.'
```

Comment:
Would be cool if we could have a more specific jsonPointer here that points to the tariff name that is a duplicate. (E.g. `/priceInfo/2/name` if the 3rd tariff's name is a duplicate of an earlier one)

Also nice would be to include multiple schemaErrors if there are multiple duplicates, so the integrator doesn't need to fix them one by one.

Reply:
Should be fixed now